File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/feature_scaling.py
# -*- coding: utf-8 -*-
"""
Created on Wed Jun 14 11:55:30 2017

@author: pradeep
"""

## Data Preprocessing

## Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

## Importing the dataset
dataset = pd.read_csv('C:\Pradeep\Project\Learning\ML A-Z\P1 Data Preprocessing\Machine-Learning-A-Z\Part 1 - Data Preprocessing\Data.csv')
X = dataset.iloc[:, :-1].values
y = dataset.iloc[:, 3].values

## Taking care of missing data
from sklearn.preprocessing import Imputer
imputer = Imputer(missing_values = 'NaN', strategy = 'mean', axis = 0)
imputer = imputer.fit(X[:, 1:3])
X[:, 1:3] = imputer.transform(X[:, 1:3])


# Coverting categorical data to numerical value 

from sklearn.preprocessing import LabelEncoder,OneHotEncoder

labelencoder_X=LabelEncoder()

X[:,0]=labelencoder_X.fit_transform(X[:,0]) 

##array([0, 2, 1, 2, 1, 0, 2, 0, 1, 0]
##0=France, 2=Spain, 1=Germany

oneHotEncoder=OneHotEncoder(categorical_features=[0]) #categorical_features encode First column

X=oneHotEncoder.fit_transform(X).toarray()

labelencoder_y=LabelEncoder()

y=labelencoder_X.fit_transform(y)   #Since it is a dependent variable. Label encoder is enough


#Splitting the data in to training and test set
from sklearn.cross_validation import train_test_split

#20% of the data will be test set (good choice will be, 20 or 25%)
X_train,X_test,y_train,y_test=train_test_split(X,y,test_size=0.2,random_state=0)


#feature scaling
from sklearn.preprocessing import StandardScaler
scaling_X=StandardScaler()
X_train=scaling_X.fit_transform(X_train)
X_test=scaling_X.transform(X_test)

File: Part 1 - Data Preprocessing/Section 2 -------------------- Part 1 - Data Preprocessing --------------------/feature_scaling.R

#**Import the data set**

#**set the working directory**

setwd("C:/Pradeep/Project/Learning/ML A-Z/P1 Data Preprocessing/Machine-Learning-A-Z/Part 1 - Data Preprocessing")


#**Import the data set**

dataset=read.csv("data.csv")


#**print the data set**

print(dataset)

#**Matrix creation is not required in R**


## Take care of missing data

dataset$Age = ifelse(is.na(dataset$Age),
                     ave((dataset$Age),FUN = function(x) mean(x,na.rm = TRUE))
                     ,dataset$Age)



dataset$Salary = ifelse(is.na(dataset$Salary),
                        ave((dataset$Salary),FUN = function(x) mean(x,na.rm = TRUE))
                        ,dataset$Salary)


## Encoding the categorical data

dataset$Country=factor(dataset$Country,levels = c("France","Spain","Germany"),labels = c(1,2,3))

print(dataset$Country)

dataset$Purchased=factor(dataset$Purchased,levels = c("Yes","No"),labels = c(0,1))

print(dataset$Purchased)


#Splitting the data in to training and test set
#install a catools
#install.packages("caTools")

library(caTools)

set.seed(123)

split = sample.split(dataset$Purchased,SplitRatio = 0.2)
test_set=subset(dataset,split=TRUE)
training_Set=subset(dataset,split=FALSE)



# Feature Scaling
 training_set[, 2:3] = scale(training_set[, 2:3])
 test_set[, 2:3] = scale(test_set[, 2:3])